packages/miniflare/src/plugins/assets/index.ts

```diff
@@ -196,9 +212,26 @@ const walk = async (dir: string) => {
 					);
 				}
 
-				manifest.push(
-					await hashPath(encodeFilePath(relativeFilepath, path.sep))
+				// Each manifest entry is [header, pathHash, contentHash].
+				// In dev we want to avoid reading and hashing every file every time we
+				// reload the dev server, so in dev the 'content hash' is hash(path + modifiedTime).
+				// This way the 'content hash' still updates on file change, so the eTag also updates,
+				// and we don't incorrectly get a 304.
+				const pathHash = await hashPath(
```

Comment:
```suggestion
				/*
				 * Each manifest entry is a series of bytes that store data in a [header,
				 * pathHash, contentHash] format, where:
				 *   - `header` is a fixed ??x?? bytes reserved for ??
				 *   - `pathHash` is the hashed file path
				 *   - `contentHash` is the hashed file content
				 *
				 * The `contentHash` of a file is determined by reading the contents of
				 * the file, and applying a hash function on the read data. In local
				 * development, performing this operation for each asset file would
				 * become very expensive very quickly, as it would have to be performed
				 * every time a dev server reload is trigerred. In watch mode, depending
				 * on the user's setup, this could potentially be on every file change.
				 *
				 * To avoid this from becoming a performance bottleneck, we're doing
				 * things a bit differently for dev, and implementing the `contentHash`
				 * as a hash function of the file path and the modified timestamp.
				 * reload the dev server, so in dev the 'content hash' is hash(path + modifiedTime).
				 * This way a file's corresponding 'contentHash' will always update
				 * if the file changes, and `wrangler dev` will serve the updated asset
				 * files instead of incorrectly returning 304s.
				 */
```